### PR TITLE
Move perf_info into AP_Scheduler

### PR DESCRIPTION
**Pull request by peterbarker**
Shares code between Copter and Sub.

Hopefully will be able to move Plane and Rover to the same structure.

### CODE REVIEW

**Comment by khancyr** [comment]
LGTM!
Nice, I will be able to use this instead of https://github.com/ArduPilot/ardupilot/pull/6828/commits/1926e9f5008e492c49f299253365638e7b8127d2

**Comment by peterbarker** [comment]
Thanks @amilcarlucas and @khancyr 

This PR was a stepping-stone for #7223  - a somewhat more ambitious patch set.

File: libraries/AP_Scheduler/PerfInfo.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    uint32_t get_min_time() const;
+    uint16_t get_num_long_running() const;
+    uint32_t get_num_dropped() const;
+    uint32_t get_avg_time() const;
+    uint32_t get_stddev_time() const;
+
+private:
+
+    // 400hz loop update rate
+#define OVERTIME_THRESHOLD_MICROS 3000
```
Comment (OXINARF) [inline libraries/AP_Scheduler/PerfInfo.h]: Please move to cpp file, doesn't look to be needed anywhere else.

File: libraries/AP_Scheduler/PerfInfo.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    uint16_t loop_count;
+    uint32_t max_time; // in microseconds
+    uint32_t min_time; // in microseconds
+    uint64_t sigma_time;
+    uint64_t sigmasquared_time;
+    uint16_t long_running;
+    uint32_t log_dropped;
+    bool ignore_loop = false;
```
Comment (OXINARF) [inline libraries/AP_Scheduler/PerfInfo.h]: Initialization not needed?

File: libraries/AP_Scheduler/PerfInfo.h (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Scheduler/PerfInfo.h]: Done

File: libraries/AP_Scheduler/PerfInfo.h (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Scheduler/PerfInfo.h]: Fixed.

**Comment by peterbarker** [comment]
@OXINARF The `X+Y: message` thing was me testing the waters.  Particularly for a strict file-move like this I think it's worthwhile :-)

**Comment by peterbarker** [comment]
@OXINARF I've pushed up create() method changes.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**